orchestra/static/orchestra/project/js/controllers.js

```diff
@@ -0,0 +1,870 @@
+            tasksVis.distribute();
+          });
+
+          tasksVis.distribute();
+        };
+
+        tasksVis.drawBackgrounds = function() {
+          d3.selectAll('.task-name').transition()
+            .style({
+              'background-color': function(slug, i) {
+                return i % 2 == 0 ? '#eee' : 'white'
+              }
+            })
+          d3.selectAll('.task-view').transition()
+            .style({
+              'background-color': function(slug, i) {
+                return i % 2 == 0 ? '#eee' : 'white'
+              }
+            })
+        }
+
+        tasksVis.drawRevertLines = function() {
+          var tasks = vis.parentContainer.selectAll('.task');
+          var revertGroups = tasks.selectAll('.revert-group').data(function(slug) {
+            var task = dataService.taskFromKey(slug);
+            var datetimes = [];
+            _.each(task.assignments, function(assignment) {
+              if (assignment.iterations) {
+                _.each(assignment.iterations, function(iteration) {
+                  datetimes.push({
+                    'datetime': iteration.start_datetime.toDate(),
+                    'taskKey': task.step_slug
+                  });
+                });
+              }
+            })
+            if (!dataService.inProgressAssignment(task)) {
+              datetimes.push({
+                'datetime': dataService.taskEnd(task).toDate(),
+                'taskKey': task.step_slug
+              });
+            }
+            return datetimes;
+          }, function(datetime) {return datetime.datetime});
+          revertGroups.exit().remove();
+          var revertGroupsEnter = revertGroups.enter().append('g')
+            .attr('class', 'revert-group');
+
+          revertGroupsEnter.append('line')
+            .attr({
+              'class': 'revert-line',
+              'stroke': 'rgb(0, 121, 191)',
+            })
+
+          // Revert flags
+          revertGroupsEnter.append('path')
+            .attr({
+              'd': 'M0,0 V4 L-2,2 Z',
+              'fill': 'rgb(0, 121, 191)',
+              'transform': vis.utils.translateString(0, -vis.options.lanePadding.top / 2) + ' scale(3, 3)',
+            })
+            .style('cursor', 'pointer')
+            .on('mouseenter', function(datetimeInfo) {
+              if (!vis.reverting) {
+                vis.crosshair.move(datetimeInfo.datetime);
+                vis.crosshair.show();
+              }
+            })
+            .on('mouseleave', function(datetimeInfo) {
+              if (!vis.reverting) {
+                vis.crosshair.hide();
+              }
+            })
+            .on('click', function(datetimeInfo) {
+              if (vis.reverting) {
+                return;
+              }
+              vis.reverting = true;
+              var taskId = dataService.taskFromKey(datetimeInfo.taskKey).id;
+              orchestraApi.revert(taskId, datetimeInfo.datetime, 'fake')
```

Comment:
i you split this out, you might implement it in the angular code and pass in a callback/function to the projectVis that facilitates this